packages/core/core/src/migrations/draft-publish.ts

```diff
@@ -31,22 +33,37 @@ const enableDraftAndPublish = async ({ oldContentTypes, contentTypes }: Input) =
       const oldContentType = oldContentTypes[uid];
       const contentType = contentTypes[uid];
 
+      const isLocalized = strapi
+        .plugin('i18n')
+        .service('content-types')
+        .isLocalizedContentType(oldContentType);
```

Comment:
I wasn't sure whether we wanted to target the old CT or the new one since I'm not sure about what we're trying to achieve exactly. 

cc @Marc-Roig for confirmation